releasenotes/notes/bool_expression_phaseoracle-1802be3016c83fa8.yaml

```diff
@@ -0,0 +1,97 @@
+---
+features:
+  - |
+    New gate-like class 
+    :class:`qiskit.circuit.classicalfunction.boolean_expression.BooleanExpression`
+    allows creating an oracles a from a Python boolean expression.
+
+    .. jupyter-execute::
+
+      from qiskit.circuit import BooleanExpression, QuantumCircuit
+      
+      expression = BooleanExpression('~x & (y | z)')
+      circuit = QuantumCircuit(4)
+      circuit.append(expression, [0, 1, 2, 3])
+      circuit.draw('mpl')
+
+    .. jupyter-execute::
+
+      circuit.decompose().draw('mpl')
+
+  - |
+    The new method
+    :meth:`qiskit.circuit.classicalfunction.boolean_expression.BooleanExpression.from_dimacs_file`
+    allows loading formulas described in the DIMACS-CNF format.  
+
+    .. code-block::
+
+      from qiskit.circuit import BooleanExpression, QuantumCircuit
+      boolean_exp = BooleanExpression.from_dimacs_file("simple_v3_c2.cnf")
+      circuit = QuantumCircuit(boolean_exp.num_qubits)
+      circuit.append(boolean_exp, range(boolean_exp.num_qubits))
+      circuit.draw('text')
+           ┌───────────────────┐
+      q_0: ┤0                  ├
+           │                   │
+      q_1: ┤1                  ├
+           │  SIMPLE_V3_C2.CNF │
+      q_2: ┤2                  ├
+           │                   │
+      q_3: ┤3                  ├
+           └───────────────────┘
+
+      circuit.decompose().draw('text')
+      q_0: ──o────o────────────
+             │    │            
+      q_1: ──■────o────■───────
+             │    │    │       
+      q_2: ──■────┼────o────■──
+           ┌─┴─┐┌─┴─┐┌─┴─┐┌─┴─┐
+      q_3: ┤ X ├┤ X ├┤ X ├┤ X ├
+           └───┘└───┘└───┘└───┘
+
+  - |
+    The new class
+    :class:`qiskit.circuit.library.phase_oracle.PhaseOracle`
+    allows to construct phase oracle circuits from Python boolean expressions.
+
+    .. jupyter-execute::
+
+      from qiskit.circuit.library.phase_oracle import PhaseOracle
+      oracle = PhaseOracle('x1 & x2 & (not x3)')
+      oracle.draw('mpl')
+    
+    These phase oracles can be used, for example, in 
+    :class:`qiskit.algorithms.amplitude_amplifiers.amplification_problem.AmplificationProblem`.
+
+    .. jupyter-execute::
+
+      from qiskit.algorithms.amplitude_amplifiers.amplification_problem import AmplificationProblem
+      from qiskit.algorithms.amplitude_amplifiers.grover import Grover
+      from qiskit import BasicAer
+      
+      backend = BasicAer.get_backend('qasm_simulator')
+      
+      problem = AmplificationProblem(oracle, is_good_state=oracle.evaluate_bitstring)
+      grover = Grover(quantum_instance=backend)
+      result = grover.amplify(problem)
+      result.top_measurement
+
+  - |
+    The new method
+    :meth:`qiskit.algorithms.amplitude_amplifiers.amplification_problem.AmplificationProblem.from_dimacs_file`
```

Comment:
Hm I can't find this method, am I missing something or has this not yet been included? 🙂

Reply:
Fixed in 8988e1c36